app/models/concerns/user_concern.rb

```diff
@@ -172,6 +172,13 @@ def can_access_project?(project, permission: :can_view_projects)
       cached_viewable_project_ids(permission: permission).include?(project.id)
     end
 
+    def can_view_my_dashboard?
+      Rails.cache.fetch('user/can_view_my_dashboard', expires_in: 1.minutes) do
+        # todo @martha - possible to manually expire cache when user impersonation changes?
```

Comment:
The cache key should contain the user id, since we only want to cache the result per user -- we don't want to cache it globally, which would return the same thing for everyone.

I like the approach taken elsewhere in this file: `key = [self.class.name, __method__, id]`. The first 2 items are to uniquely specify this use case –– its unlikely that we'll accidentally collide with some cache key elsewhere in the application. The last item is the user id, which ensures that each user will have their values cached separately.

Once the `id` is in the cache key, you shouldn't need to do anything special for impersonation. When impersonating someone else, the `current_user` will have a different `id`, so this will re-evaluate. Good thought to test that!